R/f7List.R

```diff
@@ -1,106 +1,110 @@
+#'         f7List(
+#'           mode = "media",
+#'           lapply(1:3, function(j) {
+#'             f7ListItem(
+#'               title = letters[j],
+#'               subtitle = "subtitle",
+#'               "Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 #'             Nulla sagittis tellus ut turpis condimentum, ut dignissim
 #'             lacus tincidunt. Cras dolor metus, ultrices condimentum sodales
 #'             sit amet, pharetra sodales eros. Phasellus vel felis tellus.
 #'             Mauris rutrum ligula nec dapibus feugiat. In vel dui laoreet,
 #'             commodo augue id, pulvinar lacus.",
-#'             media = tags$img(
-#'              src = paste0(
-#'              "https://cdn.framework7.io/placeholder/people-160x160-", j, ".jpg"
-#'              )
-#'             ),
-#'             right = "Right Text"
-#'           )
-#'         })
-#'       ),
+#'               media = tags$img(
+#'                 src = paste0(
+#'                   "https://cdn.framework7.io/placeholder/people-160x160-", j, ".jpg"
+#'                 )
+#'               ),
+#'               right = "Right Text"
+#'             )
+#'           })
+#'         ),
 #'
-#'       # list with links
-#'       f7List(
-#'         lapply(1:3, function(j) {
-#'           f7ListItem(url = "https://google.com", letters[j])
-#'         })
-#'       ),
+#'         # list with links
+#'         f7List(
+#'           mode = "links",
+#'           lapply(1:3, function(j) {
+#'             tags$li(
+#'               f7Link(label = letters[j], href = "https://google.com")
+#'             )
+#'           })
+#'         ),
 #'
-#'       # grouped lists
-#'       f7List(
-#'         mode = "contacts",
-#'         lapply(1:3, function(i) {
-#'           f7ListGroup(
-#'             title = LETTERS[i],
-#'             lapply(1:3, function(j) f7ListItem(letters[j]))
-#'           )
-#'         })
+#'         # grouped lists
+#'         f7List(
+#'           mode = "contacts",
+#'           lapply(1:3, function(i) {
+#'             f7ListGroup(
+#'               title = LETTERS[i],
+#'               lapply(1:3, function(j) f7ListItem(letters[j]))
+#'             )
+#'           })
+#'         )
 #'       )
-#'     )
-#'   ),
-#'   server = function(input, output) {}
-#'  )
+#'     ),
+#'     server = function(input, output) {}
+#'   )
 #' }
-f7List <- function(..., mode = NULL, inset = FALSE) {
-
+f7List <- function(
+    ..., mode = NULL, inset = FALSE, outline = FALSE,
+    dividers = FALSE, strong = FALSE) {
   listCl <- "list chevron-center"
-  if (!is.null(mode)) listCl <- paste0(listCl, " ", mode, "-list")
-  if (inset) listCl <- paste0(listCl, " inset")
+  if (strong) listCl <- paste(listCl, "list-strong")
+  if (outline) listCl <- paste(listCl, "list-outline")
+  if (dividers) listCl <- paste(listCl, "list-dividers")
+  if (!is.null(mode)) listCl <- paste(listCl, sprintf("%s-list", mode))
+  if (inset) listCl <- paste(listCl, "inset")
```

Comment:
@DivadNojnarg this should also be implemented in `f7VirtualList`, I forgot that. Now the class is `list` only but strong, outline, dividers, inset can also be added. Copy paste I think

Reply:
I'll leave a note in the task list.